picard/ui/searchdialog.py

```diff
@@ -0,0 +1,466 @@
+        self.setSelectionBehavior(
+                QtGui.QAbstractItemView.SelectRows)
+        self.setEditTriggers(
+                QtGui.QAbstractItemView.NoEditTriggers)
+        self.horizontalHeader().setStretchLastSection(True)
+        self.horizontalHeader().setResizeMode(
+                QtGui.QHeaderView.Stretch)
+        self.horizontalHeader().setResizeMode(
+                QtGui.QHeaderView.Interactive)
+
+class SearchBox(QtGui.QWidget):
+
+    def __init__(self, parent):
+        self.parent = parent
+        QtGui.QWidget.__init__(self, parent)
+        self.search_action = QtGui.QAction(icontheme.lookup('system-search'),
+                _(u"Search"), self)
+        self.search_action.triggered.connect(self.search)
+        self.setupUi()
+
+    def setupUi(self):
+        self.layout = QtGui.QVBoxLayout(self)
+        self.search_row_widget = QtGui.QWidget(self)
+        self.search_row_layout = QtGui.QHBoxLayout(self.search_row_widget)
+        self.search_row_layout.setContentsMargins(1, 1, 1, 1)
+        self.search_row_layout.setSpacing(1)
+        self.search_edit = ButtonLineEdit(self.search_row_widget)
+        self.search_row_layout.addWidget(self.search_edit)
+        self.search_button = QtGui.QToolButton(self.search_row_widget)
+        self.search_button.setAutoRaise(True)
+        self.search_button.setDefaultAction(self.search_action)
+        self.search_button.setIconSize(QtCore.QSize(22, 22))
+        self.search_row_layout.addWidget(self.search_button)
+        self.search_row_widget.setLayout(self.search_row_layout)
+        self.layout.addWidget(self.search_row_widget)
+        self.adv_opt_row_widget = QtGui.QWidget(self)
+        self.adv_opt_row_layout = QtGui.QHBoxLayout(self.adv_opt_row_widget)
+        self.adv_opt_row_layout.setAlignment(QtCore.Qt.AlignLeft)
+        self.adv_opt_row_layout.setContentsMargins(1, 1, 1, 1)
+        self.adv_opt_row_layout.setSpacing(1)
+        self.use_adv_search_syntax = QtGui.QCheckBox(self.adv_opt_row_widget)
+        self.use_adv_search_syntax.setText(_("Use advance query syntax"))
+        self.adv_opt_row_layout.addWidget(self.use_adv_search_syntax)
+        self.adv_syntax_help = QtGui.QLabel(self.adv_opt_row_widget)
+        self.adv_syntax_help.setOpenExternalLinks(True)
+        self.adv_syntax_help.setText(_(
+                "&#160;(<a href='https://musicbrainz.org/doc/Indexed_Search_Syntax'>"
+                "Syntax Help</a>)"))
+        self.adv_opt_row_layout.addWidget(self.adv_syntax_help)
+        self.adv_opt_row_widget.setLayout(self.adv_opt_row_layout)
+        self.layout.addWidget(self.adv_opt_row_widget)
+        self.layout.setContentsMargins(1, 1, 1, 1)
+        self.layout.setSpacing(1)
+        self.setMaximumHeight(60)
+
+    def search(self):
+        self.parent.search(self.search_edit.text())
+
+    def restore_checkbox_state(self):
+        self.use_adv_search_syntax.setChecked(config.setting["use_adv_search_syntax"])
```

Comment:
Whoops, sorry, I didn't realize this is an already existing options (I think I've never used one of the search functions other than Lookup in Picard ;-) ). Ignore the above comment